src/cpio_cache.rs

```diff
@@ -219,8 +373,8 @@ pub enum CpioError {
     #[error("An IO error")]
     Io {
         ctx: &'static str,
-        src: PathBuf,
-        dest: PathBuf,
+        src: Option<PathBuf>,
+        dest: Option<PathBuf>,
         e: std::io::Error,
```

Comment:
This line should have `#[source]` so error reporters can show the error chain. 
```suggestion
        #[source]
        e: std::io::Error,
```